bundles/org.openhab.binding.pirateweather/src/main/java/org/openhab/binding/pirateweather/internal/handler/PirateWeatherWeatherAndForecastHandler.java

```diff
@@ -0,0 +1,867 @@
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelGroupTypeUID;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link PirateWeatherWeatherAndForecastHandler} is responsible for handling commands, which are sent to one of the
+ * channels.
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+@NonNullByDefault
+public class PirateWeatherWeatherAndForecastHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PirateWeatherWeatherAndForecastHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections
+            .singleton(THING_TYPE_WEATHER_AND_FORECAST);
+
+    private static final String PRECIP_TYPE_SNOW = "snow";
+    private static final String PRECIP_TYPE_RAIN = "rain";
+
+    private static final String CHANNEL_GROUP_HOURLY_FORECAST_PREFIX = "forecast-hours-";
+    private static final String CHANNEL_GROUP_DAILY_FORECAST_PREFIX = "forecast-day-";
+    private static final String CHANNEL_GROUP_ALERTS_PREFIX = "alerts-";
+    private static final Pattern CHANNEL_GROUP_HOURLY_FORECAST_PREFIX_PATTERN = Pattern
+            .compile(CHANNEL_GROUP_HOURLY_FORECAST_PREFIX + "([0-9]*)");
+    private static final Pattern CHANNEL_GROUP_DAILY_FORECAST_PREFIX_PATTERN = Pattern
+            .compile(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + "([0-9]*)");
+    private static final Pattern CHANNEL_GROUP_ALERTS_PREFIX_PATTERN = Pattern
+            .compile(CHANNEL_GROUP_ALERTS_PREFIX + "([0-9]*)");
+
+    // keeps track of all jobs
+    private static final Map<String, Job> JOBS = new ConcurrentHashMap<>();
+
+    // keeps track of the parsed location
+    protected @Nullable PointType location;
+    // keeps track of the parsed counts
+    private int forecastHours = 24;
+    private int forecastDays = 8;
+    private int numberOfAlerts = 0;
+
+    private @Nullable PirateWeatherChannelConfiguration sunriseTriggerChannelConfig;
+    private @Nullable PirateWeatherChannelConfiguration sunsetTriggerChannelConfig;
+    private @Nullable PirateWeatherJsonWeatherData weatherData;
+
+    public PirateWeatherWeatherAndForecastHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
```

Comment:
Minor comment, code can be reduced and the method can be simplified when the channel rebuilding logic is extracted to a private helper method.

Reply:
I mean the part around lines 160-230 should be extracted